src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -236,6 +236,9 @@ private async Task InnerOnConnectionAsync(ConnectionContext context)
                 await using (sslStream)
                 {
                     await _next(context);
```

Comment:
If this throws, do we ever dispose the SslDuplexPipe?

Reply:
we could do:

await using (sslStream)
await using (sslDuplexPipe)
{
     await _next(context);
}